programs/mango-v4/tests/cases/test_health_compute.rs

```diff
@@ -142,6 +169,145 @@ async fn test_health_compute_tokens_during_maint_weight_shift() -> Result<(), Tr
+        })
+        .collect();
+
+    // let fallback_metas = vec![];
+
+    //
+    // SETUP: Create a group and an account
+    //
+
+    let GroupWithTokens { group, tokens, .. } = GroupWithTokensConfig {
+        admin,
+        payer,
+        mints: mints.to_vec(),
+        ..GroupWithTokensConfig::default()
+    }
+    .create(solana)
+    .await;
+
+    let account =
+        create_funded_account(&solana, group, owner, 0, &context.users[1], &[], 1000, 0).await;
+
+    let mut cu_measurements = vec![];
+    for token_account in &context.users[0].token_accounts[..mints.len()] {
+        deposit_cu_datapoint(solana, account, owner, *token_account).await;
+    }
+
+    //
+    // SETUP: Create and register fallback oracles for each token
+    //
+    for (i, _token_account) in context.users[0].token_accounts[..mints.len()]
+        .iter()
+        .enumerate()
+    {
+        send_tx(
+            solana,
+            StubOracleCreate {
+                oracle: fallback_oracle_kps[i],
+                group,
+                mint: mints[i].pubkey,
+                admin,
+                payer,
+            },
+        )
+        .await
+        .unwrap();
+
+        send_tx(
+            solana,
+            TokenEdit {
+                group,
+                admin,
+                mint: mints[i].pubkey,
+                fallback_oracle: fallback_oracle_kps[i].pubkey(),
+                options: mango_v4::instruction::TokenEdit {
+                    set_fallback_oracle: true,
+                    ..token_edit_instruction_default()
+                },
+            },
+        )
+        .await
+        .unwrap();
+    }
+
+    //
+    // TEST: Progressively make each oracle invalid so that the fallback is used
+    //
+    for (i, token_account) in context.users[0].token_accounts[..mints.len()]
+        .iter()
+        .enumerate()
+    {
+        send_tx(
+            solana,
+            StubOracleSetTestInstruction {
+                oracle: tokens[i].oracle,
+                group,
+                mint: mints[i].pubkey,
+                admin,
+                price: 1.0,
+                last_update_slot: 0,
+                deviation: 100.0,
+            },
+        )
+        .await
+        .unwrap();
+
+        cu_measurements.push(
+            deposit_cu_fallbacks_datapoint(
+                solana,
+                account,
+                owner,
+                *token_account,
+                fallback_metas.clone(),
+            )
+            .await,
+        );
+    }
+
+    for (i, pair) in cu_measurements.windows(2).enumerate() {
+        println!(
+            "after adding token {}: {} (+{})",
+            i,
+            pair[1],
+            pair[1] - pair[0]
+        );
+    }
+
+    let avg_cu_increase = cu_measurements.windows(2).map(|p| p[1] - p[0]).sum::<u64>()
+        / (cu_measurements.len() - 1) as u64;
+    println!("average cu increase: {avg_cu_increase}");
+    assert!(avg_cu_increase < 16_600);
```

Comment:
wow, it shouldn't be _that_ expensive, I'll also take a look; let's try to figure out what's eating compute!